web/src/main/java/com/bazaarvoice/emodb/web/system/SystemSource.java

```diff
@@ -0,0 +1,15 @@
+package com.bazaarvoice.emodb.web.system;
+
+import java.util.Map;
+
+public interface SystemSource {
```

Comment:
There's a missing implementation of this.  The two implementations provided either update the state in the local data center or call to a remote instance and update it's state.  What needs to happen is both.  For example, when a Stash starts in us-east-1 it needs to update the local state in us-east-1 AND make a call to eu-west-1 and update its state using the resource endpoint.  Unless both take place there is no guarantee that compaction would be restricted while Stash is running.